### DISCUSSION TITLE
Cluster unable to update in 4.15.0-0.okd-2024-02-10-035534

### DISCUSSION THREAD

**Discussion by llomgui** [okd-project/okd#1884] [General]
Hello,

I tried to update a cluster to the latest version `4.15.0-0.okd-2024-02-10-035534`.
It looks like rpm-ostree does not want to use proxy variables set in `/etc/mco/proxy.env`.

The master node keeps failing on `rpm-ostree rebase --experimental ostree-unverified-registry:quay.io/openshift/okd-content@sha256:114f56ffe30b010ee62acc1744e195954e7194cc816b8fc0a23802fa048a50df` > Timeout issue on quay.io.
It is not using the proxy as it should.

I also tried the following without luck:
`mkdir -p /etc/systemd/system/rpm-ostreed.service.d
cat > /etc/systemd/system/rpm-ostreed.service.d/http-proxy.conf << EOF
[Service]
Environment="http_proxy=my-proxy:8080"
EOF`

The following commands work:

`export HTTP_PROXY=my-proxy:8080
podman pull quay.io/openshift/okd-content@sha256:114f56ffe30b010ee62acc1744e195954e7194cc816b8fc0a23802fa048a50df`

**Comment by vrutkovs** [comment]
Please attach (or upload to the public file sharing service) [must-gather](https://docs.okd.io/latest/support/gathering-cluster-data.html#support_gathering_data_gathering-cluster-data) archive

**Reply by llomgui** [reply]
I cannot share must-gather logs due to company policy.

**Reply by vrutkovs** [reply]
I don't know how we can help then

**Reply by llomgui** [reply]
I created the same issue 1 year ago: https://github.com/okd-project/okd/issues/1481
But this time the solution is not working.

**Reply by nate-duke** [reply]
@llomgui you can try using the redhatter's behind-the-scenes tool [o-must-gather](https://github.com/kxr/o-must-gather). I've not used this myself but if your company doesn't allow you to share logs with the community then that may be your best bet.

**Reply by llomgui** [reply]
Thank you @nate-duke If Vadim forces me I will.

But I found something: https://github.com/ostreedev/ostree-rs-ext/issues/582

It looks like they reset all env variable in the latest ostree version. It might be my issue here @vrutkovs

**Reply by llomgui** [reply]
And their hack creating a file `/usr/local/bin/skopeo` works 👍

**Reply by vrutkovs** [reply]
Lets track this in https://github.com/okd-project/okd/issues/1481

**Comment by tonus-sebastian** [comment]
anyone give how to step by step upgrade cluster

**Reply by llomgui** [reply]
See my comment: https://github.com/okd-project/okd/issues/1481#issuecomment-1962862849
